### PR TITLE
WIP - Filter documentation

### PR DESCRIPTION
**Pull request by MarkGibbons**
### Description

Document the sample filters

### Issues Resolved

Need working examples

### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

---
This change is [Reviewable](https://reviewable.io/reviews/sous-chefs/line/142)

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big Pull Request.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by MarkGibbons** [comment]
Still working on this one.

File: documentation/resources/filters/stanza.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Examples for the stanza filter
+
+## Original file
+````
+[first]
+line1 = value1
```
Comment (JohnRoesler) [inline documentation/resources/filters/stanza.md]: to match the formatting, should all lines under the bracketed headings be indented? line 8, 15, & 18 also

File: resources/filter_lines.rb [hunk 2]
```diff
@@ -45,7 +45,7 @@
   # filter ::= <code> | { <code> => <args> }
   # args ::= <String> | <Array>
   # code ::= <Symbol> | <Method> | <Proc>
-  # Symbol ::= :after | :before | :between | :comment | :missing | :replace | :stanza | :substitute
+  # Symbol ::= :after | :before | :between | :comment | :delete_between | :missing | :replace | :stanza | :substitute
```
Comment (JohnRoesler) [inline resources/filter_lines.rb:47]: Test failing on this file:
```
    TypeError
    ---------
    no implicit conversion of nil into String
    
    Cookbook Trace:
    ---------------
    /opt/kitchen/cache/cookbooks/line/resources/filter_lines.rb:34:in `exist?'
    /opt/kitchen/cache/cookbooks/line/resources/filter_lines.rb:34:in `block in class_from_file'
```

Comment (MarkGibbons) [reply resources/filter_lines.rb:47]: Very much still work in progress.  The stanza filter has to be merged before this one.  Thanks  for looking at it.

File: documentation/resources/filters/stanza.md (hunk 1 above)
Comment (MarkGibbons) [reply documentation/resources/filters/stanza.md]: I'll make it prettier and line them up.  The inserted lines won't necessarily match the existing format.

File: resources/filter_lines.rb (hunk 2 above)
Comment (MarkGibbons) [reply resources/filter_lines.rb:47]: Should be fixed now.

**Review by JohnRoesler — DISMISSED** [review]

**Review by JohnRoesler — APPROVED** [review]

**Pull request merged**